test/LibraryManager.IntegrationTest/LibmanCompletionTests.cs

```diff
@@ -33,5 +32,152 @@ public void LibmanCompletion_Destination()
+
+            string[] expectedCompletionEntries = new[] {
+                "cdnjs",
+                "filesystem",
+                "unpkg",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown();
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_DefaultDestination()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "Properties/",
+                "wwwroot/",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown();
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultDestination\":");
+
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_LibraryForCdnjs()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "bootstrap",
+                "jquery",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"cdnjs\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("j");
+            LibmanTestsUtility.WaitForCompletionEntryNotPresent(Editor, "bootstrap", caseInsensitive: true);
+
+            Editor.KeyboardCommands.Type("q");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        [Ignore("Ignored for bug 628945")]
+        public void LibmanCompletion_LibraryVersionForCsnjs()
+        {
+            _libManConfig.Open();
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"cdnjs\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("jquery@");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, new string[] { }, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_LibraryForFilesystem()
```

Comment:
This test is just verifying completion filtering, which is not implemented in nor unique to LibMan.  I don't think we should test it here.

Reply:
I suppose it is verifying the URL picker, but hte last two bits are just filtering and should be removed.

Reply:
Will do

---
In reply to: [193926181](https://github.com/aspnet/LibraryManager/pull/108#discussion_r193926181) [](ancestors = 193926181)

Reply:
I noticed that the PathCompletionProvider implemented in the libman solution will handle this completion request. So I think it would be useful to test here, right?

---
In reply to: [193925927](https://github.com/aspnet/LibraryManager/pull/108#discussion_r193925927) [](ancestors = 193925927)